ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/SignedBlsToExecutionChangeValidator.java

```diff
@@ -13,32 +13,130 @@
 import tech.pegasys.teku.spec.datastructures.operations.SignedBlsToExecutionChange;
 import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
 import tech.pegasys.teku.spec.logic.common.operations.validation.OperationInvalidReason;
+import tech.pegasys.teku.storage.client.RecentChainData;
 
 public class SignedBlsToExecutionChangeValidator
     implements OperationValidator<SignedBlsToExecutionChange> {
 
-  // FIXME Implement validation logic (#6358)
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Spec spec;
+  private final RecentChainData recentChainData;
+  private final Set<UInt64> seenBlsToExecutionChangeMessageFromValidators =
+      LimitedSet.createSynchronized(VALID_VALIDATOR_SET_SIZE);
+
+  public SignedBlsToExecutionChangeValidator(final Spec spec, RecentChainData recentChainData) {
+    this.spec = spec;
+    this.recentChainData = recentChainData;
+  }
 
   @Override
   public SafeFuture<InternalValidationResult> validateFully(
       final SignedBlsToExecutionChange operation) {
-    return SafeFuture.completedFuture(InternalValidationResult.ACCEPT);
+
+    final BlsToExecutionChange blsToExecutionChange = operation.getMessage();
+    final UInt64 validatorIndex = blsToExecutionChange.getValidatorIndex();
+
+    /*
+     [IGNORE] The signed_bls_to_execution_change is the first valid signed bls to execution change received for the
+     validator with index signed_bls_to_execution_change.message.validator_index.
+    */
+    if (!isFirstBlsToExecutionChangeForValidator(blsToExecutionChange)) {
+      final String logMessage =
+          String.format(
+              "BlsToExecutionChange is not the first one for validator %s.", validatorIndex);
+      LOG.trace(logMessage);
+      return SafeFuture.completedFuture(InternalValidationResult.create(IGNORE, logMessage));
+    }
+
+    /*
+     [REJECT] All of the conditions within process_bls_to_execution_change pass validation.
+    */
+    return getFailureReason(operation)
+        .thenApply(
+            failureReason -> {
+              if (failureReason.isPresent()) {
+                return InternalValidationResult.reject(
+                    "Exit for validator %s is invalid: %s",
+                    validatorIndex, failureReason.get().describe());
+              }
+
+              if (seenBlsToExecutionChangeMessageFromValidators.add(validatorIndex)) {
+                return InternalValidationResult.ACCEPT;
+              } else {
+                final String msg =
+                    String.format(
+                        "SignedBlsToExecutionChangeValidator: operation is not the first one for validator %s.",
+                        validatorIndex);
+                LOG.trace(msg);
+                return InternalValidationResult.create(IGNORE, msg);
+              }
+            });
   }
 
   @Override
   public Optional<OperationInvalidReason> validateForStateTransition(
```

Comment:
We kinda don't need this method. It will be removed in a followup PR. Just kept it to be consistent with the interface before the refactor.

Reply:
i thought this was for light validation when applying blocks... i think when it becomes useful is not going to be for a while, so we could potentially remove it for now... i agree..